tensorflow/python/training/learning_rate_decay_test.py

```diff
@@ -108,5 +110,75 @@ def testPiecewiseConstantEdgeCases(self):
         pc = learning_rate_decay.piecewise_constant(x, boundaries, values)
 
 
+class ExponentialDecayTest(test_util.TensorFlowTestCase):
+
+  def testDecay(self):
+    initial_lr = 0.1
+    k = 10
+    decay_rate = 0.96
+    with self.test_session():
+      step = state_ops.variable_op([], dtypes.int32)
+      for i in range(k+1):
+        assign = state_ops.assign(step, i)
```

Comment:
to make it clear, it is best practice to create the entire graph ahead of time and then invoke it by calling .eval() or sess.run().

This is adding a new op (assign) for every iteration of the loop.

What about moving this statement out of the loop and changing it to assign_add(step, 1) instead ?